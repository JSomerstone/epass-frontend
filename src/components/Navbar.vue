<template>
    <b-navbar>
        <template slot="brand">
          <b-navbar-item tag="router-link" :to="{ path: '/' }">
            <b-icon icon="passport" size="is-large"></b-icon>
            ePass
          </b-navbar-item>
        </template>
        <template slot="start">
            <b-navbar-dropdown label="Tournaments">
                <b-navbar-item 
                  v-for="year in years"
                  v-bind:key="year"
                  tag="router-link" 
                  :to="{ path: `/tournaments/${year}` }"
                >
                    {{ year }}
                </b-navbar-item>
            </b-navbar-dropdown>
        </template>

        <template slot="end">
            <b-navbar-item tag="div">
                <div class="buttons" v-if="!isLoggedIn">
                  <router-link :to="{ path: '/login'}" class="button is-primary">
                    <strong>Login</strong>
                  </router-link>
                  <router-link :to="{ path: '/signup'}" class="button is-light">
                    Sign up
                  </router-link>
                </div>
                <b-navbar-dropdown  v-else :label="username">
                  <b-navbar-item to="/settings" tag="router-link">
                      <b-icon icon="account-cog-outline"></b-icon> Settings
                  </b-navbar-item>
                  <b-navbar-item @click="handleLogout">
                      <b-icon icon="logout-variant"></b-icon> Logout
                  </b-navbar-item>
              </b-navbar-dropdown>
            </b-navbar-item>
        </template>
    </b-navbar>
</template>
<style>
.navbar-brand {
  font-size: x-large;
}
.navbar-item > .icon {
  padding-right: 1em;
}
</style>
<script>
export default {
  data() {
    return {
      years: [2020, 2019]
    }
  },
  computed: {
    isLoggedIn() {
      return this.$store.getters['auth/loggedIn'];
    },
    getCurrent() {
      return this.$store.getters['auth/user'];
    },
    username: function() {
      return this.getCurrent.email.split("@").shift()
    }
  },
  methods: {
    handleLogout: function() {
      this.$store.dispatch("auth/logout");
    }
  },
}
</script>