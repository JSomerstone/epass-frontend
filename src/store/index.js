import Vue from "vue";
import Vuex from "vuex";

import axios from "axios";
import config from "../config/index";

import tournaments from "./modules/tournaments";
import referees from "./modules/referees";
import countries from "./modules/countries";
import auth from "./modules/auth";

axios.defaults.baseURL = config.backendUrl;
Vue.use(Vuex);

const store = new Vuex.Store({
  modules: {
    tournaments,
    referees,
    countries,
    auth
  }
});

export default store;