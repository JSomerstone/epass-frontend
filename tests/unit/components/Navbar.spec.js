import { mount, RouterLinkStub } from "@vue/test-utils";
import Vuex from "vuex";
import localVue from "../localVue";
import store from "../mockStore";
import testReferees from "../../data/referee.json";

import Navbar from "@/components/Navbar";

describe("Navbar", () => {

  let { auth, referees } = store.modules;


  beforeEach(() => {
    auth.state.userInfo = { };
    referees.state.current = {};
  });

  it("Renders Login & Signup buttons when not logged in", () => {

    const vuexStore = new Vuex.Store(store);
    const wrapper = mount(Navbar, {
      localVue,
      store: vuexStore,
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("a.login-link").text()).toBe("Login");
    expect(wrapper.find("a.sign-up-link").text()).toBe("Sign up");
    let link = wrapper.findComponent(RouterLinkStub);
    expect(link.props().to.path).toBe('/')
  });

  it("Does not render Login & Signup buttons when logged in", () => {
    auth.state.userInfo = {
      userId: "bogus-user",
      refereeId: testReferees[0].id,
    }
    const vuexStore = new Vuex.Store(store);
    const wrapper = mount(Navbar, {
      localVue,
      store: vuexStore,
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("a.login-link").exists()).toBe(false);
    expect(wrapper.find("a.sign-up-link").exists()).toBe(false);
    // "firtsName lastName" of the first item in test/data/referees.json
    expect(wrapper.find("a.settings-link").text()).toBe("Settings");
    expect(wrapper.find("a.logout-link").text()).toBe("Logout");
  });


  it("Calls auth/logout when Logout button is clicked", async () => {
    auth.state.userInfo = {
      userId: "bogus-user",
      refereeId: testReferees[0].id,
    }
    const logout = jest.fn();
    auth.actions = { logout }

    const vuexStore = new Vuex.Store(store);
    const wrapper = mount(Navbar, {
      localVue,
      store: vuexStore,
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    const logoutBtn = wrapper.find("a.logout-link");
    await logoutBtn.trigger('click');
    expect(logout).toHaveBeenCalled();
  });
});