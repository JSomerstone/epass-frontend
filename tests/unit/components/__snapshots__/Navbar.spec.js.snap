// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Does not render Login & Signup buttons when logged in 1`] = `
<nav role="navigation" aria-label="main navigation" class="navbar">
  <div class="navbar-brand"><a class="navbar-item"><span class="icon is-large"><i class="mdi mdi-passport mdi-48px"></i></span>
      ePass
    </a><a role="button" aria-label="menu" class="navbar-burger burger"><span aria-hidden="true"></span> <span aria-hidden="true"></span> <span aria-hidden="true"></span></a></div>
  <div class="navbar-menu">
    <div class="navbar-start">
      <div class="navbar-item has-dropdown"><a role="menuitem" aria-haspopup="true" href="#" class="navbar-link">Tournaments</a>
        <div class="navbar-dropdown"><a class="navbar-item"><span class="icon"><i class="mdi mdi-plus mdi-24px"></i></span>New
          </a> <a class="navbar-item">
            2020
          </a><a class="navbar-item">
            2019
          </a></div>
      </div> <a class="navbar-item"><a>
          Statistics
        </a></a>
    </div>
    <div class="navbar-end">
      <div class="navbar-item">
        <div class="navbar-item has-dropdown is-hoverable"><a role="menuitem" aria-haspopup="true" href="#" class="navbar-link">John Doe</a>
          <div class="navbar-dropdown"><a class="navbar-item settings-link"><span class="icon"><i class="mdi mdi-account-cog-outline mdi-24px"></i></span> Settings
            </a> <a class="navbar-item logout-link"><span class="icon"><i class="mdi mdi-logout-variant mdi-24px"></i></span> Logout
            </a></div>
        </div>
      </div>
    </div>
  </div>
</nav>
`;

exports[`Navbar Renders Login & Signup buttons when not logged in 1`] = `
<nav role="navigation" aria-label="main navigation" class="navbar">
  <div class="navbar-brand"><a class="navbar-item"><span class="icon is-large"><i class="mdi mdi-passport mdi-48px"></i></span>
      ePass
    </a><a role="button" aria-label="menu" class="navbar-burger burger"><span aria-hidden="true"></span> <span aria-hidden="true"></span> <span aria-hidden="true"></span></a></div>
  <div class="navbar-menu">
    <div class="navbar-start">
      <!----> <a class="navbar-item"><a>
          About
        </a></a></div>
    <div class="navbar-end">
      <div class="navbar-item">
        <div class="buttons"><a class="button is-primary login-link"><strong>Login</strong></a> <a class="button is-light sign-up-link">
            Sign up
          </a></div>
      </div>
    </div>
  </div>
</nav>
`;
